### PR TITLE
[Paddle]Add chapter_recurrent-neural-networks

### PR DESCRIPTION
**Pull request by tngt**

### CODE REVIEW

**Comment by w5688414** [comment]
Hi, @AnirudhDagar @cheungdaven chapter recurrent neural networks is ready, can you review this chapter? Thanks

File: chapter_recurrent-neural-networks/rnn-scratch.md [hunk 1]
```diff
@@ -644,6 +768,45 @@ def train_epoch_ch8(net, train_iter, loss, updater, use_random_iter):
     return math.exp(metric[0] / metric[1]), metric[1] / timer.stop()
 ```
 
+```{.python .input}
+#@tab paddle
+#@save
+def train_epoch_ch8(net, train_iter, loss, updater, device, use_random_iter):
+    """训练网络一个迭代周期（定义见第8章"""
```
Comment (cheungdaven) [inline chapter_recurrent-neural-networks/rnn-scratch.md:775]: missing ")"

**Review by cheungdaven — CHANGES_REQUESTED** [review]
please fix the conflicts. Thanks.

File: chapter_recurrent-neural-networks/rnn-scratch.md [hunk 2]
```diff
@@ -644,6 +768,45 @@ def train_epoch_ch8(net, train_iter, loss, updater, use_random_iter):
...
+    for X, Y in train_iter:
+        if state is None or use_random_iter:
+            # 在第一次迭代或使用随机抽样时初始化`state`
+            state = net.begin_state(batch_size=X.shape[0])
+        else:
+            if isinstance(net, nn.Layer) and not isinstance(state, tuple):
+                # `state`对于`nn.GRU`是个张量
+                state.stop_gradient=True
+            else:
+                # `state`对于`nn.LSTM`或对于我们从零开始实现的模型是个张量
```
Comment (cheungdaven) [inline chapter_recurrent-neural-networks/rnn-scratch.md:787]: remove ``

**Review by cheungdaven — CHANGES_REQUESTED** [review]

**Comment by cheungdaven** [comment]
Also, the 困惑度 of paddle is too high  (in rnn-scratch and rnn-concise).

**Comment by tngt** [comment]
> Also, the 困惑度 of paddle is too high (in rnn-scratch and rnn-concise).

@w5688414 could you help me evaluate this problem？thanks！！！

**Pull request merged**